4-projects/business_unit_1/shared/example_infra_pipeline.tf

```diff
@@ -62,6 +68,70 @@ module "infra_pipelines" {
   private_worker_pool_id      = local.cloud_build_private_worker_pool_id
 }
 
+resource "google_service_account" "workload_sa" {
```

Comment:
qual a service account que le a imagem, a que est fazendo o deploy da build ou a do workload?
se for a da build ela já tem permissão se for a do workload ela deveria ter permissão no artifact registry que já existe